### ISSUE TITLE
Enable custom user agent string in header

### ISSUE THREAD

**Issue by juzo** [appspector/flutter-plugin#50]
Is it possible to set a custom user agent in header after Appspector is inited?

Currently I am setting my own user agent string with

`IOClient(HttpClient()..userAgent = '$kAppName/$kAppBuild')`

But after 

`AppSpectorPlugin.run(config);`

My user agent string is overwritten with the default one: Dart/2.14 (dart:io)

The list of Monitors is set as:

`config.monitors = <Monitor>[Monitors.logs, Monitors.http];`

**Comment by deszip** [comment]
Hi @juzo , thanks for reporting this. Can you please tell on what platform are you seeing this issue, iOS or Android? And what version of plugin do you use?
Thanks.

**Comment by juzo** [comment]
Hey @deszip
Issues is present on both platforms Android and iOS.

Using the plugin version 0.8.0.

Thanks.

**Comment by Dimdron** [comment]
have you used the previous version?

**Comment by juzo** [comment]
Yes, the issue was present with older versions of the plugin.

I believe the problem lies in setting the `HttpOverride` globals in

```
  AppSpectorPlugin._withConfig(Config config, Function(String)? sessionUrlListener) {
    HttpOverrides.global = AppSpectorHttpOverrides();
    _requestReceiver.observeChannel();
    _channel.setMethodCallHandler(_handlePluginCalls);
    _sessionUrlListener = sessionUrlListener;
    _appSpectorPlugin = this;
  }
```

If we could pass an optional user agent to `AppSpectorHttpClient` class this issue could be easily solved.

**Comment by Dimdron** [comment]
@juzo hey! The fix is included to the 0.8.1 version

**Comment by juzo** [comment]
Great! Thanks @Dimdron
